controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +124,296 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 	return result, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Consumer Mode")
+func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim controller in Consumer Mode")
+
+	providerClient, err := providerclient.NewProviderClient(
+		r.ctx,
+		r.storageCluster.Spec.ExternalStorage.StorageProviderEndpoint,
+		10*time.Second,
+	)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Close client-side connections.
+	defer providerClient.Close()
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+
+		// Check if finalizers are present, if not, add them.
+		if !contains(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer) {
+			storageClassClaimRef := klog.KRef(r.storageClassClaim.Name, r.storageClassClaim.Namespace)
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", storageClassClaimRef)
+			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), StorageClassClaimFinalizer))
+			if err := r.update(r.storageClassClaim); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to update StorageClassClaim [%v] with finalizer: %s", storageClassClaimRef, err)
+			}
+		}
+
+		// Creation phase.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimCreating
```

Comment:
Not all claims are newly created. Shouldn't you first check for what phase the claim is in, before assuming it is "creating"?

Reply:
The way I see it is that creating phase is the phase where the operator ensures that everything that need to be created is created or create it on the spot. It is not about the actual act of creating resources. 

As such I don't see an issue with setting this to creating

Reply:
The way I see it, when you reconcile a resource because it was updated, reporting it's status as "creating" is confusing/misleading. It could be "reconciling", "progressing", "updating".

I am fine with "creating" for now. We can rethink after we get user feedback.

Reply:
So let's ditch the "creating" and go directly to "configuring"/"updating"